.gitallowed

```diff
@@ -0,0 +1,32 @@
+# Each line is a regular expression against the output of git-secrets
```

Comment:
The risk of allow-listing files is that somebody could accidentally add a secret to one of the files, and it wouldn't be caught. Don't know if I'm being overly paranoid, but throwing it out there for consideration.

Reply:
Yeah, it's a razor's edge of not wanting to run people off with a bunch of false positives. It seemed like these these files relied on strings that were key-like, those strings might change with version bumps, and the line number might also change. It's potentially more secure to mark just the key-like string. I can do that, but I expect other developers to have to maintain the `.gitallowed` more often then.

Reply:
I must admit in the past, I think I've been paranoid enough to not have a `.gitallowed` and just use `--no-verify` or similar on specific commits when I know I'm modifying something suspicious (which isn't often).

Reply:
Was this file auto-generated? Would it be possible to auto-generate? This file doesn't seem easily maintainable. Also it isn't clear what the reason is for these exceptions.

Reply:
I generated the file by hand by looking at the results of git-secrets. The first version was an allow all on the files. This was seen as too permissive in review so I added the key-like string that was causing a hit for each of the files. This attempts to be a balance of allowing those strings to move line numbers and having to constantly deal with results, and always detecting if a new key-like string was added to a file.

I don't know of an easy way to autogenerate it, but I tried to leave some helpful comments. We could perhaps document it further in the wiki.

I expect it to be something like, "added an s3 object key to a file > failed commit due to git secrets > add a line to gitallowed for the file and key-like string". The risk of leaving in old data is very low. I expect these types of changes to be pretty rare (usually in setting up testing apparatuses).

It's certainly possible to devise a script that will automatically generate the .gitallowed lines and add it to the gitallowed. I'm not sure exactly how to set it up without it making it too easy to circumvent and if managing this file gets unwieldy we might have to look into it. Fortunately it didn't take terribly long to generate the file for the whole project using the output of `git secrets -scan`.

Reply:
Okay, if we expect the data to change rarely then I'm ok with an artisinal hand-crafted list of warnings to ignore. :)